File: Tordo/backend-Tordo/Models/Contabilidad/cuentas.cs
namespace backend_Tordo.Models.Contabilidad
{
  public class cuentas
  {

    public class Activo
    {
      public int Id { get; set; }
      public string Categoria { get; set; }
      public string Descripcion { get; set; }
      public decimal Monto { get; set; }
      public DateTime FechaAdquisicion { get; set; }
      public DateTime? FechaValoracion { get; set; }
      public int? VidaUtilEstimada { get; set; }
    }

    public class Pasivo
    {
      public int Id { get; set; }
      public string Categoria { get; set; }
      public string Descripcion { get; set; }
      public decimal Monto { get; set; }
      public DateTime FechaIncidencia { get; set; }
      public DateTime FechaVencimiento { get; set; }
    }

    public class Patrimonio
    {
      public int Id { get; set; }
      public string Tipo { get; set; }
      public string Descripcion { get; set; }
      public decimal Monto { get; set; }
      public DateTime FechaRegistro { get; set; }
    }

    public class Ingreso
    {
      public int Id { get; set; }
      public string Categoria { get; set; }
      public string Descripcion { get; set; }
      public decimal Monto { get; set; }
      public DateTime FechaRegistro { get; set; }
      public string Fuente { get; set; }
      public int CuentaRelacionadaId { get; set; }
    }

    public class Egreso
    {
      public int Id { get; set; }
      public string Categoria { get; set; }
      public string Descripcion { get; set; }
      public decimal Monto { get; set; }
      public DateTime FechaRegistro { get; set; }
      public string Destinatario { get; set; }
      public int CuentaRelacionadaId { get; set; }
    }

  }
}

File: Tordo/src/app/Interfaz/cuentas.ts
export interface Activo {
    id: number;
    categoria: string;
    descripcion: string;
    monto: number;
    fechaAdquisicion: Date;
    fechaValoracion?: Date;
    vidaUtilEstimada?: number;
}

export interface Pasivo {
    id: number;
    categoria: string;
    descripcion: string;
    monto: number;
    fechaIncidencia: Date;
    fechaVencimiento: Date;
}

export interface Patrimonio {
    id: number;
    categoria: string;
    tipo: string;
    descripcion: string;
    monto: number;
    fechaRegistro: Date;
}

export interface Ingreso {
    id: number;
    categoria: string;
    descripcion: string;
    monto: number;
    fechaRegistro: Date;
    fuente: string;
    cuentaRelacionadaId: number;
}

export interface Egreso {
    id: number;
    categoria: string;
    descripcion: string;
    monto: number;
    fechaRegistro: Date;
    destinatario: string;
    cuentaRelacionadaId: number;
}

export interface EstructuraDto {
    tipo: string;
    niveles: number;
    subniveles: number[];
  }
  
  export interface Estructura {
    id: number;
    nombre: string;
    tipo: string;
    nivel: number;
    subniveles?: Estructura[]; // Subniveles es opcional y puede ser una lista de estructuras
  }
  
  export interface Finanza {
    id: number;
    categoria: string;
    descripcion: string;
    monto: number;
    fecha?: Date; // Fecha general para cualquier tipo de finanza
    tipo?: string;
    Etipo?: string; // Puede ser 'Activo', 'Pasivo', 'Patrimonio', 'Ingreso', 'Egreso'
    // ... Otras propiedades específicas
}